### PR TITLE
feat: added more checks for function creation check

### PR DESCRIPTION
**Pull request by zhengbuqian**
issue: https://github.com/milvus-io/milvus/issues/35853

* BM25 Function now takes no params, k1, b should be passed via index params
* support BM25 full text search when metric type is not present in search request
* add more strict validation with functions at collection creation time

### CODE REVIEW

File: internal/proxy/task_index.go [hunk 1]
```diff
@@ -165,8 +152,15 @@ func (cit *createIndexTask) parseFunctionParamsToIndex(indexParamsMap map[string
 		if _, ok := indexParamsMap["bm25_avgdl"]; !ok {
 			indexParamsMap["bm25_avgdl"] = "100"
 		}
+
+		if metricType, ok := indexParamsMap["metric_type"]; !ok {
```
Comment (czs007) [inline internal/proxy/task_index.go:156]: what if metric_type is empty "" ?

Comment (zhengbuqian) [reply internal/proxy/task_index.go:156]: it goes to the error branch.

File: internal/proxy/task_index.go [hunk 2]
```diff
@@ -165,8 +152,15 @@ func (cit *createIndexTask) parseFunctionParamsToIndex(indexParamsMap map[string
...
+			indexParamsMap["metric_type"] = "BM25"
+		} else if metricType != "BM25" {
```
Comment (czs007) [inline internal/proxy/task_index.go:158]: what if metric_type is empty "" ?

Comment (zhengbuqian) [reply internal/proxy/task_index.go:158]: providing empty metric type at index creation time leads to error and will fail the request

**Comment by zhengbuqian** [comment]
rerun go-sdk

**Pull request merged**